### PR TITLE
Merge simple changes from miami into master

### PR DESCRIPTION
**Pull request by Philpax**
This PR improves maintainability by bringing over changes from `miami` that do not impact how the game functions. There are a few changes that might be a little controversial, but I figured they were like that in `miami` for a reason. Will happily remove any particular change for approval.

### CODE REVIEW

File: src/text/Messages.cpp [hunk 1]
```diff
@@ -419,10 +420,9 @@ CMessages::InsertStringInString(wchar *str1, wchar *str2)
 	for (i = 0; i < total_size; ) {
 #ifdef MORE_LANGUAGES
 		if ((CFont::IsJapanese() && *_str1 == (0x8000 | '~') && *(_str1 + 1) == (0x8000 | 'a') && *(_str1 + 2) == (0x8000 | '~'))
-			|| (*_str1 == '~' && *(_str1 + 1) == 'a' && *(_str1 + 2) == '~'))
-		{
+			|| (*_str1 == '~' && *(_str1 + 1) == 'a' && *(_str1 + 2) == '~')) {
 #else
-			if (*_str1 == '~' && *(_str1 + 1) == 'a' && *(_str1 + 2) == '~') {
+		if (*_str1 == '~' && *(_str1 + 1) == 'a' && *(_str1 + 2) == '~') {
```
Comment (madebr) [inline src/text/Messages.cpp:425]: I think this indentation is incorrect.

File: src/rw/MemoryHeap.cpp [hunk 2]
```diff
@@ -7,6 +7,8 @@
 #include "FileLoader.h"
 #include "MemoryHeap.h"
 
+// TODO(MIAMI)
+
 #ifdef USE_CUSTOM_ALLOCATOR
```
Comment (madebr) [inline src/rw/MemoryHeap.cpp:12]:
```suggestion
#ifdef USE_CUSTOM_ALLOCATOR
```

File: src/rw/MemoryHeap.cpp [hunk 3]
```diff
@@ -339,7 +341,7 @@ CMemoryHeap::TidyHeap(void)
 	}
 }
 
-// 
+// MIAMI: this is empty
```
Comment (madebr) [inline src/rw/MemoryHeap.cpp:344]:
```suggestion
//
```

**Comment by Sergeanur** [comment]
These are VC changes that belong to miami only. We would only take bug fixes from VC, and those should be wrapped into `#ifdef FIX_BUGS` too.

**Pull request closed without merging**